library/mobjects.c

```diff
@@ -69,21 +69,23 @@ int eblob_base_setup_data(struct eblob_base_ctl *ctl, int force)
 	struct stat st;
 	int err;
 
-	err = fstat(ctl->index_fd, &st);
-	if (err) {
-		err = -errno;
-		goto err_out_exit;
+	if (ctl->index_ctl.fd >= 0) {
```

Comment:
eblob_base_ctl_open() calls eblob_base_setup_data() before initializing the index.